tests/src/JIT/SIMD/VectorCast.cs

```diff
@@ -0,0 +1,193 @@
+        succeeded &= Test_SByte  <UInt32> ();
+        succeeded &= Test_UInt16 <UInt32> ();
+        succeeded &= Test_Int16  <UInt32> ();
+        succeeded &= Test_UInt32 <UInt32> ();
+        succeeded &= Test_Int32  <UInt32> ();
+        succeeded &= Test_UInt64 <UInt32> ();
+        succeeded &= Test_Int64  <UInt32> ();
+        succeeded &= Test_Single <UInt32> ();
+        succeeded &= Test_Double <UInt32> ();
+                                 
+        succeeded &= Test_Byte   <Int32>  ();
+        succeeded &= Test_SByte  <Int32>  ();
+        succeeded &= Test_UInt16 <Int32>  ();
+        succeeded &= Test_Int16  <Int32>  ();
+        succeeded &= Test_UInt32 <Int32>  ();
+        succeeded &= Test_Int32  <Int32>  ();
+        succeeded &= Test_UInt64 <Int32>  ();
+        succeeded &= Test_Int64  <Int32>  ();
+        succeeded &= Test_Single <Int32>  ();
+        succeeded &= Test_Double <Int32>  ();
+                                 
+        succeeded &= Test_Byte   <UInt64> ();
+        succeeded &= Test_SByte  <UInt64> ();
+        succeeded &= Test_UInt16 <UInt64> ();
+        succeeded &= Test_Int16  <UInt64> ();
+        succeeded &= Test_UInt32 <UInt64> ();
+        succeeded &= Test_Int32  <UInt64> ();
+        succeeded &= Test_UInt64 <UInt64> ();
+        succeeded &= Test_Int64  <UInt64> ();
+        succeeded &= Test_Single <UInt64> ();
+        succeeded &= Test_Double <UInt64> ();
+                                 
+        succeeded &= Test_Byte   <Int64>  ();
+        succeeded &= Test_SByte  <Int64>  ();
+        succeeded &= Test_UInt16 <Int64>  ();
+        succeeded &= Test_Int16  <Int64>  ();
+        succeeded &= Test_UInt32 <Int64>  ();
+        succeeded &= Test_Int32  <Int64>  ();
+        succeeded &= Test_UInt64 <Int64>  ();
+        succeeded &= Test_Int64  <Int64>  ();
+        succeeded &= Test_Single <Int64>  ();
+        succeeded &= Test_Double <Int64>  ();
+                                 
+        succeeded &= Test_Byte   <Single> ();
+        succeeded &= Test_SByte  <Single> ();
+        succeeded &= Test_UInt16 <Single> ();
+        succeeded &= Test_Int16  <Single> ();
+        succeeded &= Test_UInt32 <Single> ();
+        succeeded &= Test_Int32  <Single> ();
+        succeeded &= Test_UInt64 <Single> ();
+        succeeded &= Test_Int64  <Single> ();
+        succeeded &= Test_Single <Single> ();
+        succeeded &= Test_Double <Single> ();
+                                          
+        succeeded &= Test_Byte   <Double> ();
+        succeeded &= Test_SByte  <Double> ();
+        succeeded &= Test_UInt16 <Double> ();
+        succeeded &= Test_Int16  <Double> ();
+        succeeded &= Test_UInt32 <Double> ();
+        succeeded &= Test_Int32  <Double> ();
+        succeeded &= Test_UInt64 <Double> ();
+        succeeded &= Test_Int64  <Double> ();
+        succeeded &= Test_Single <Double> ();
+        succeeded &= Test_Double <Double> ();
```

Comment:
Yep, that's a common test problem. I've changed the code to report the line number. Seems to be a good compromise:
* the test author doesn't have to generate a bunch of numbers manually
* it's trivial from a codegen point of view compared to other possible solutions (e.g. using reflection to print out test method names).